### DISCUSSION TITLE
What do default values for columns actually do?

### DISCUSSION THREAD

**Discussion by campbellcole** [SeaQL/sea-orm#2217] [Q&A]
I've been trying to understand what `#[sea_orm(default_value = ...)]` and `ColumnDef::default_value()` actually do to the generated objects. Since the model must have every field, at what point is this default value used? Is it used when an ActiveModel with the corresponding column = `ActiveValue::Unset` gets inserted?

Thanks in advance.

**Comment by tyt2y3** [comment]
It's the corresponding database default expression, like `default current_timestamp`, or `default 1`.
It's not related to the ActiveModel.

**Reply by campbellcole** [reply]
So is the database responsible for providing this value? If I'm understanding correctly, it doesn't actually do anything to the code but it signals to the developer that the database will use this value if it is omitted in an ActiveModel? Thank you for the reply.

**Reply by detoxifiedplant** [reply]
Yes, i've been trying to understand it's use case, after trying around with couple of methods, it seems it actually does not make any changes to the query.
is it @tyt2y3 ?

**Reply by tyt2y3** [reply]
It's only used in [create_table_from_entity](https://docs.rs/sea-orm/latest/sea_orm/schema/struct.Schema.html#method.create_table_from_entity) which affects the generated schema of the table.

If you want to specify default value on client side for Active Models, see https://github.com/SeaQL/sea-orm/blob/1.1.x/tests/common/bakery_chain/cake.rs#L54 for an example.

I'll try to update the docs to clarify this.

**Reply by detoxifiedplant** [reply]
Thanks, things are way easier now.
